File: src/androidTest/java/com/simprints/libsimprints/RefusalFormTest.java
package com.simprints.libsimprints;

import android.os.Parcel;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import junit.framework.Assert;

import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
public class RefusalFormTest {

    @Test
    public void testRefusalFormParcelling() {
        RefusalForm example = new RefusalForm("No Scanner", "Forgot Scanner");

        Parcel parcel = Parcel.obtain();
        example.writeToParcel(parcel, 0);

        parcel.setDataPosition(0);

        RefusalForm createdFromParcel = RefusalForm.CREATOR.createFromParcel(parcel);
        Assert.assertEquals(example, createdFromParcel);
    }
}

File: src/test/java/com/simprints/libsimprints/RefusalFormTest.kt
package com.simprints.libsimprints;

import androidx.core.os.bundleOf
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class RefusalFormTest {

    @Test
    fun `test refusal form parcelling`() {
        val expected = RefusalForm("No Scanner", "Forgot Scanner")
        val actual = bundleOf("test" to expected).getParcelable<RefusalForm>("test")

        assertEquals(expected, actual)
    }
}
